simulators/active_surface.py

```diff
@@ -91,53 +133,244 @@ def get_status(self):
-            self.resolution = self.resolutions.get(0)
+        self.stop_io_enable[2] = int(binary_string[5], 2)
+        if self.stop_io_enable[2] == 1:
+            self.stop_io_level[2] = int(binary_string[2], 2)
         else:
-            self.auto_resolution = False
-            self.resolution = self.resolutions.get(int(param[-3:], base=2))
+            self.stop_io_level[2] = 0
+
+    def set_positioning_io(self, param):
+        binary_string = bin(param)[2:].zfill(8)
+
+        # binary_string[0 - 1] is currently unused
+
+        self.pos_io_enable[0] = int(binary_string[7], 2)
+        if self.pos_io_enable[0] == 1:
+            self.pos_io_level[0] = int(binary_string[4], 2)
+        else:
+            self.pos_io_level[0] = 0
+
+        self.pos_io_enable[1] = int(binary_string[6], 2)
+        if self.pos_io_enable[1] == 1:
+            self.pos_io_level[1] = int(binary_string[3], 2)
+        else:
+            self.pos_io_level[1] = 0
+
+        self.pos_io_enable[2] = int(binary_string[5], 2)
+        if self.pos_io_enable[2] == 1:
+            self.pos_io_level[2] = int(binary_string[2], 2)
+        else:
+            self.pos_io_level[2] = 0
 
-    def move_to(self, new_position):
+    def set_home_io(self, param):
+        binary_string = bin(param)[2:].zfill(8)
+
+        self.home_io_enable[0] = int(binary_string[7], 2)
+        if self.home_io_enable[0] == 1:
+            self.home_io_level[0] = int(binary_string[4], 2)
+        else:
+            self.home_io_level[0] = 0
+
+        self.home_io_enable[1] = int(binary_string[6], 2)
+        if self.home_io_enable[1] == 1:
+            self.home_io_level[1] = int(binary_string[3], 2)
+        else:
+            self.home_io_level[1] = 0
+
+        self.home_io_enable[2] = int(binary_string[5], 2)
+        if self.home_io_enable[2] == 1:
+            self.home_io_level[2] = int(binary_string[2], 2)
+        else:
+            self.home_io_level[2] = 0
+
+    def set_working_mode(self, params):
+        binary_string = bin(params[0])[2:].zfill(8)
+
+        # binary_string[0 - 6] is currently unused
+        self.baud_rate = self.baud_rates.get(int(binary_string[7], 2))
+
+        #  params[1] is currently unused
+
+    def _accel_ramp(self, sign):
+        # Should gradually accelerate from 0 to min_frequency to max_frequency
+        while False:  # Not yet implemented
+            delta = 0
+            self.current_position += delta * sign
+
+    def _decel_ramp(self, sign):
+        # Should gradually decelerate from max_frequency to min_frequency to 0
+        while False:  # Not yet implemented
+            delta = 0
+            self.current_position += delta * sign
+
+    def _move_to(self, new_position):
+        # Should be launched in a separate thread
         # ((freq / resolution) / 200)*60 = RPM
-        self.current_position = new_position
+
+        self.running = True
+
+        if new_position < self.current_position:
```

Comment:
You can also write it more compact, without losing readability, using the so called **ternary expression**:

```
sign = -1 if new_position < self.current_position else +1
```